src/libasr/pass/replace_symbolic.cpp

```diff
@@ -294,6 +291,51 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
         pass_result.push_back(al, stmt);
     }
 
+    void process_binary_operator(Allocator &al,  const Location &loc, ASR::IntrinsicFunction_t* x, SymbolTable* module_scope,
+        const std::string& new_name, ASR::expr_t* target) {
+            ASR::expr_t* value1 = x->m_args[0];
+            ASR::expr_t* value2 = x->m_args[1];
+
+            if (ASR::is_a<ASR::IntrinsicFunction_t>(*value1) || ASR::is_a<ASR::Cast_t>(*value1)) {
+                if (ASR::is_a<ASR::IntrinsicFunction_t>(*value1)) {
+                    this->visit_IntrinsicFunction(*ASR::down_cast<ASR::IntrinsicFunction_t>(value1));
+                } else {
+                    this->visit_Cast(*ASR::down_cast<ASR::Cast_t>(value1));
+                }
```

Comment:
This lines seems to be repeated as well, so we will create a new function say: `handle_arguments ` or something similar to handle it.
```c
ASR::expr_t* value1 = handle_arguments(x->m_args[0]);
```

Reply:
Inside the function just check for `CPtr` type (for other `type` throw Exception), `Var_t`, `IntrinsicFunction_t` and `Cast_t` ( for other `expr_t` throw Exception).